streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -74,24 +77,29 @@
     private final MockProducer<byte[], byte[]> nonEosMockProducer = new MockProducer<>(
         cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer);
     private final StreamsProducer nonEosStreamsProducer =
-        new StreamsProducer(nonEosMockProducer, false, logContext);
+        new StreamsProducer(nonEosMockProducer, AT_LEAST_ONCE, logContext);
 
-    private final MockProducer<byte[], byte[]> eosMockProducer = new MockProducer<>(
+    private final MockProducer<byte[], byte[]> eosAlphaMockProducer = new MockProducer<>(
         cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer);
-    private final StreamsProducer eosStreamsProducer =
-        new StreamsProducer(eosMockProducer, true, logContext);
+    private final StreamsProducer eosAlphaStreamsProducer =
+        new StreamsProducer(eosAlphaMockProducer, EXACTLY_ONCE_ALPHA, logContext);
+
+    private final MockProducer<byte[], byte[]> eosBetaMockProducer = new MockProducer<>(
+        cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer);
+    private final StreamsProducer eosBetaStreamsProducer =
+        new StreamsProducer(eosBetaMockProducer, EXACTLY_ONCE_BETA, logContext);
 
     private final ProducerRecord<byte[], byte[]> record =
         new ProducerRecord<>(topic, 0, 0L, new byte[0], new byte[0], new RecordHeaders());
 
     @Before
     public void before() {
-        eosStreamsProducer.initTransaction();
+        eosAlphaStreamsProducer.initTransaction();
     }
 
 
 
-    // generic tests (non-EOS and EOS)
+    // generic tests (non-EOS and EOS-alpha/beta)
```

Comment:
I don't think that works, as the `StreamsProducer` has different behavior for the three different processing modes -- it would only work if we expect the same behavior to different parameters.

Reply:
Not necessarily, we could pass in both processing mode and the expected output as parameters, if the test workflow looks essentially the same.

Reply:
Well, what I tried to say is, that the test workflow is not the same for most cases.